src/cogent3/core/new_moltype.py

```diff
@@ -716,9 +716,12 @@ def get_degenerate_positions(
                 f"{seq[:4]!r} not valid for moltype {self.name!r}"
             )
 
+        if self.degen_alphabet is None:
+            return []
+
         alpha = self.most_degen_alphabet()
         seq = alpha.to_indices(seq)
-        for index, val in enumerate(self.degen_gapped_alphabet):
+        for index, val in enumerate(alpha):
```

Comment:
Same issue as we've discussed before. You can assume that the ambiguity indices occur after the `len(moltype)+1` -- plus one for the gap index.